js/style/style_layer/line_style_layer.js

```diff
@@ -10,11 +11,9 @@ class LineStyleLayer extends StyleLayer {
         // If the line is dashed, scale the dash lengths by the line
         // width at the previous round zoom level.
         if (value && name === 'line-dasharray') {
-            const flooredZoom = Math.floor(globalProperties.zoom);
-            if (this._flooredZoom !== flooredZoom) {
-                this._flooredZoom = flooredZoom;
-                this._flooredLineWidth = this.getPaintValue('line-width', globalProperties, featureProperties);
-            }
+            const flooredGlobalProperties = util.clone(globalProperties);
+            flooredGlobalProperties.zoom = Math.floor(globalProperties.zoom);
+            this._flooredLineWidth = this.getPaintValue('line-width', flooredGlobalProperties, featureProperties);
```

Comment:
Is there a need to keep the instance variable or can this be `const flooredLineWidth = ...`?